src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -243,84 +192,81 @@ public void buildSettlement(UserBean user, String gameIdString, String nodeIdStr
 
     @Override
     public void endTurn(UserBean user, String gameIdString) throws PlayException, GameException {
-        Integer nextMoveNumber = null;
+        log.debug("User {} tries to end his turn of game id {}", user == null ? "<EMPTY>" : user.getUsername(), gameIdString);
+
         GameBean game = gameUtil.getGameById(gameIdString, ERROR_CODE_ERROR);
-        List<List<String>> initialBuildingsSet = gameUtil.getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+        validateGameStatusIsPlaying(game);
+        validateUserNotEmpty(user);
+        validateCurrentTurnOfUser(game, user);
+        validateActionIsAllowed(user, game, GameUserActions.END_TURN);
 
-        log.debug("User {} tries to end his turn of game id {}",
-                user == null ? "<EMPTY>" : user.getUsername(), gameIdString);
-        //TODO: move to common validation method
-        if (user == null) {
-            log.debug("User should not be empty");
-            throw new PlayException(ERROR_CODE_ERROR);
-        }
+        playUtil.updateNextMoveOrder(game);
+        playUtil.updateCurrentCycleBuildingNumber(game);
```

Comment:
it is not clear that I update build number of cycle, but I don't know nothing about updating cycle number, this logic should be wired and placed near each other